packages/next/src/server/app-render/create-error-handler.tsx

```diff
@@ -73,37 +81,44 @@ export function createHTMLReactServerErrorHandler(
   isNextExport: boolean,
   reactServerErrors: Map<string, DigestedError>,
   silenceLogger: boolean,
-  onReactServerRenderError: undefined | ((err: any) => void)
+  onReactServerRenderError: undefined | ((err: DigestedError) => void)
 ): RSCErrorHandler {
-  return (err: any) => {
-    // If the error already has a digest, respect the original digest,
-    // so it won't get re-generated into another new error.
-    if (!err.digest) {
```

Comment:
Move this block to after we checked for known Next.js errors and surely have a proper `Error` instance. Known Next.js errors will always have a `digest` so this is save.